src/main/java/com/kodilla/ecommercee/controller/ProductController.java

```diff
@@ -1,39 +1,116 @@
+        return ResponseEntity.ok(productDtos);
+    }
+
+    @GetMapping("/productsInCart")
+    public ResponseEntity<List<ProductsInCartDto>> getProductsInCart() {
+        List<ProductsInCart> productsInCartList = productsInCartService.getAllProductsInCart();
+        List<ProductsInCartDto> productsInCartDtos = productsInCartMapper.mapToProductsInCartDtoList(productsInCartList);
+        return ResponseEntity.ok(productsInCartDtos);
     }
 
     @GetMapping("/{productId}")
-    public ResponseEntity<ProductDto> getProduct(@PathVariable Long productId) {
-        return ResponseEntity.ok(new ProductDto());    }
+    public ResponseEntity<ProductDto> getProductById(@PathVariable Long productId) throws ProductNotFoundException {
+        Product product = productService.getProductById(productId);
+        ProductDto productDto = productMapper.mapToProductDto(product);
+        return ResponseEntity.ok(productDto);
+    }
+
+    @GetMapping("/productsInCart/{productsInCartId}")
+    public ResponseEntity<ProductsInCartDto> getProductInCartById(@PathVariable Long productsInCartId) throws ProductsInCartNotFoundExeption {
+        ProductsInCart productsInCart = productsInCartService.getProductsInCartById(productsInCartId);
+        ProductsInCartDto productsIncartDto = productsInCartMapper.mapToProductsInCartDto(productsInCart);
+        return ResponseEntity.ok(productsIncartDto);
+    }
 
     @PostMapping
-    public ResponseEntity<Void> createProduct() {
+    public ResponseEntity<Void> createProduct(@RequestBody ProductDto productDto) {
+        Product product = productMapper.mapToProduct(productDto);
+        groupService.saveGroup(product.getGroup());
+        productService.saveProduct(product);
         return ResponseEntity.ok().build();
     }
 
-    @PutMapping("/{productId}")
-    public ResponseEntity<Void> updateProduct(@PathVariable Long productId) {
+    @PostMapping("/productsInCart")
+    public ResponseEntity<Void> createProductsInCart(@RequestBody ProductsInCartDto productsInCartDto) throws ProductNotFoundException {
+        ProductsInCart productsInCart = productsInCartMapper.mapToProductsInCart(productsInCartDto);
+        productsInCartService.saveProductsInCart(productsInCart);
         return ResponseEntity.ok().build();
     }
 
+    @PutMapping("/{productId}")
+    public ResponseEntity<ProductDto> updateProduct(@PathVariable Long productId, @RequestBody ProductDto productDto) throws ProductNotFoundException {
+        Product productToUpdate = productService.getProductById(productId);
+        productToUpdate.setName(productDto.getName());
+        productToUpdate.setDescription(productDto.getDescription());
+        productToUpdate.setPrice(productDto.getPrice());
+        Product updatedProduct = productService.updateProduct(productToUpdate);
+        ProductDto updatedProductDto = productMapper.mapToProductDto(updatedProduct);
+        return ResponseEntity.ok(updatedProductDto);
```

Comment:
I'm not sure I understand why you create the `updateProduct` in this way, Let me know is it because you didn't want to create another class of `ProductDto` without some fields?